internal/kafka/internal/handlers/admin_kafka.go

```diff
@@ -169,13 +170,33 @@ func (h *adminKafkaHandler) Update(w http.ResponseWriter, r *http.Request) {
 				return false
 			}
 
+			getStatusBasedOnSuspendedParam := func(susp *bool, kafka *dbapi.KafkaRequest) string {
+				if !boolIsSet(susp) {
+					return kafka.Status
+				} else {
+					if *susp {
+						if kafka.Status != constants.KafkaRequestStatusSuspended.String() {
+							return constants.KafkaRequestStatusSuspending.String()
+						}
+					} else {
+						if kafka.Status == constants.KafkaRequestStatusSuspended.String() || kafka.Status == constants.KafkaRequestStatusSuspending.String() {
+							return constants.KafkaRequestStatusResuming.String()
+						}
+					}
```

Comment:
can we add a unit test for this piece of logic?

Reply:
adding them as we speak :)

Reply:
just seen https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1344#discussion_r993107180, thanks.